node/src/main.rs

```diff
@@ -685,12 +679,57 @@ fn create_ipfs_clients(logger: &Logger, ipfs_addresses: &Vec<String>) -> Vec<Ipf
         .collect()
 }
 
-fn start_block_ingestor(
+fn networks_as_chains(
     logger: &Logger,
-    block_polling_interval: Duration,
+    node_id: NodeId,
+    registry: Arc<MetricsRegistry>,
     eth_networks: &EthereumNetworks,
-    block_store: Arc<DieselBlockStore>,
+    store: &Store,
+    chain_head_update_listener: Arc<ChainHeadUpdateListener>,
     logger_factory: &LoggerFactory,
+) -> HashMap<String, Arc<ethereum::Chain>> {
```

Comment:
How networks are going to be handled is something I was out of ideas on. Having separate `Chain` instances for different networks seems like a good solution! We will have to see how to adapt the config file format to handle different chains.

Reply:
Yes, parsing the config and the subgraph manifest will require some amount of `serde` trickery; for configs there will need to be some sort of `enum` that lists all possible chains; it's not clear to me how that could become generic code.